src/java.base/share/classes/java/io/File.java

```diff
@@ -2110,6 +2110,13 @@ static File generateFile(String prefix, String suffix, File dir)
      * is invoked, but programmatic changes to this property are not guaranteed
      * to have any effect upon the temporary directory used by this method.
      *
+     * <p> If the {@code directory} argument is not {@code null}, then if its
+     * abstract pathname is valid and denotes an existing directory, then the
+     * file will be created in that directory; otherwise, the file will not
+     * be created and an {@code IOException} will be thrown.  In no case will
```

Comment:
I might suggest  reworking the 1st sentence as it is a tad long and we could probably reduce the 'then' usages.